### ISSUE TITLE
ArgumentNullException when using both PropertyChanged.Fody and Realm

### ISSUE THREAD

**Issue by tuyen-vuduc** [realm/realm-dotnet#1758]
## Goals
I am trying to use latest PropertyChanged.Foday v2.5.8 and latest Realm v3.1.0 installed.

I am working on Xamarin project with MvvmCross and here is my weavers.xml
```
<?xml version="1.0" encoding="utf-8"?>
<Weavers>
  <PropertyChanged />
  <RealmWeaver/>
</Weavers>
```
## Expected Results
Everything just works.

## Actual Results
```
Value cannot be null.
Parameter name: instruction
System.ArgumentNullException: Value cannot be null.
Parameter name: instruction
   at Mono.Cecil.Cil.InstructionOffset..ctor(Instruction instruction)
   at Mono.Cecil.Cil.CodeReader.ReadScope(ScopeDebugInformation scope)
   at Mono.Cecil.Cil.CodeReader.ReadDebugInfo()
   at Mono.Cecil.Cil.CodeReader.ReadMethodBody()
   at Mono.Cecil.Cil.CodeReader.ReadMethodBody(MethodDefinition method)
   at Mono.Cecil.MethodDefinition.<>c.<get_Body>b__41_0(MethodDefinition method, MetadataReader reader)
   at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TRet& variable, TItem item, Func`3 read)
   at Mono.Cecil.MethodDefinition.get_Body()
   at Mono.Cecil.MethodDefinition.get_DebugInformation()
   at Mono.Debugging.Soft.SoftDebuggerSession.LoadPdbType(TypeDefinition type, Dictionary`2 fileToSourceFileInfos) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2645
   at Mono.Debugging.Soft.SoftDebuggerSession.LoadPdbFile(String assemblyFileName, String pdbFileName) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2637
   at Mono.Debugging.Soft.SoftDebuggerSession.LoadDebugFile(String assemblyFileName, String debugFileName, Func`3 loadDebugFile) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2733
   at Mono.Debugging.Soft.SoftDebuggerSession.CheckBetterMatch(TypeMirror type, String file, Int32 line, Int32 column, Location found) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2755
   at Mono.Debugging.Soft.SoftDebuggerSession.FindLocationByMethod(MethodMirror method, String file, Int32 line, Int32 column, Boolean& insideTypeRange) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2942
   at Mono.Debugging.Soft.SoftDebuggerSession.FindLocationByType(TypeMirror type, String file, Int32 line, Int32 column, Boolean& genericMethod, Boolean& insideTypeRange) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2960
   at Mono.Debugging.Soft.SoftDebuggerSession.ResolveBreakpoints(TypeMirror type) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2454
   at Mono.Debugging.Soft.SoftDebuggerSession.HandleTypeLoadEvents(TypeLoadEvent[] events) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 2008
   at Mono.Debugging.Soft.SoftDebuggerSession.HandleEventSet(EventSet es) in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 1549
   at Mono.Debugging.Soft.SoftDebuggerSession.EventHandler() in E:\A\_work\91\s\external\debugger-libs\Mono.Debugging.Soft\SoftDebuggerSession.cs:line 1471
```

## Steps to Reproduce
1. Create a Xamarin project with MvvmCross, PropertyChanged.Fody and Realm
2. Create several RealmObject classes
2. Try to debug the application

## Version of Realm and Tooling

- Realm Object Server Version: 3.1.0
- Flavor:
  - [x] Developer
  - [x] Professional
  - [ ] Enterprise
- Server OS & Version: Windows latest
- VisualStudio for Windows (no issue on VS for Mac)

**Comment by bmunkholm** [comment]
What makes you conclude this is a problem in Realm?
Could you please provide the smallest possible app that reproduces this issue and points to Realm as the rootcause?

**Comment by nielsenko** [comment]
Realm 3.1 uses Fody 3.0. See commit d0808b05

**Comment by tuyen-vuduc** [comment]
@nielsenko We are using v3.1, but it's still the issue.
@bmunkholm I will create a simple app then attach. I think it's an issue of Realm because I searched around and others faced the same and there is a guide like @nielsenko said (v3.1 fixes the issue).

**Comment by nielsenko** [comment]
@tuyen-vuduc You mention you use Fody 2.5.8. You need to upgrade to Fody 3.0.x, if you use Realm 3.1.

**Comment by nielsenko** [comment]
@tuyen-vuduc My bad. You are using PropertyChanged.Fody v. 2.5.8 that uses Fody v. 3.1.3, that you believe should be compatible with Realm 3.1, correct?

**Comment by nielsenko** [comment]
@tuyen-vuduc I have upgraded an app to use Realm 3.1 and Fody 3.1.3 without any issues, at least on iOS. However, I'm not using PropertyChange.Fody in this project.

**Comment by andrekiba** [comment]
...another time 👎  ...if inside a project you have Realm, PropertyChanged and AsyncErrorHandler you can be sure that if you update something you have problems. With Realm 3.1.0 installed I also tried to downgrade to PropertyChanged 2.5.7 and AsyncErrorHandler 1.2.0 so all the 3 have the same minimum dependence on Fody 3.0.3... but it doesn't help. Build succeds but same error at runtime. Is there a way to fix this? Please realm guys can you take in consideration to not to use PropertyChanged and implement your own weaver??

**Comment by fealebenpae** [comment]
@andrekiba Realm doesn't depend on or use PropertyChanged internally.

**Comment by andrekiba** [comment]
Sorry I meant Fody...non PropertyChanged.Fody... but, in any case, so what is the problem??

**Comment by fealebenpae** [comment]
It's difficult to say without a reproduction we can debug, but in the past we've had issues with multiple weaver packages that depended on different versions of Fody (especially with the Fody 2.x to 3.x transition).

**Comment by andrekiba** [comment]
Have the error is quite easy: just create a new project with Realm 3.1.0, PropertyChanged.Fody 2.5.9 and Fody 3.1.3

**Comment by andrekiba** [comment]
any news here? thank you

**Comment by fealebenpae** [comment]
I'm sorry, but we don't have the capacity to look into this issue right away.

**Comment by andrekiba** [comment]
ok, I think the problem is quite serious because it happend also in the past and it happens every time.
It is very common to have PropertyChanged in a Xamarin app so please take in account seriously the interaction between the Realm weaver and other weavers. I try also to open an issue on the PropertyChanged  github...if have at least an idea I try to investigate. thank you.

**Comment by tstivers1990** [comment]
I also have this issue. Using `Realm 3.4.0` and `PropertyChanged.Fody 2.6.0` (which depends on `Fody 3.3.2`). If I set a breakpoint, when the breakpoint is encountered and the debugger would normally break, an error dialog pops up in Visual Studio with the exception `ArgumentNullException`. I inherited this project and I'll be removing PropertyChanged.Fody in the near future as I don't like it, as well as replacing Realm with EF Core with SQLite provider. So I won't be waiting for a fix, but wanted to make you aware this issue still exists.
